ui/webapp.py

```diff
@@ -12,175 +13,197 @@
 
 
 def main():
-    # Define state
-    state_question = SessionState.get(
-        random_question=DEFAULT_QUESTION_AT_STARTUP, random_answer="", next_question="false", run_query="false"
+
+    # Persistent state
+    state = SessionState.get(
+        random_question=DEFAULT_QUESTION_AT_STARTUP, 
+        random_answer="",
+        results=None,
+        raw_json=None,
+        get_next_question=True
     )
 
-    # Initialize variables
-    eval_mode = False
-    random_question = DEFAULT_QUESTION_AT_STARTUP
-    eval_labels = os.getenv("EVAL_FILE", "eval_labels_example.csv")
+    # Small callback to reset the interface in case the text of the question changes
+    def reset_results(*args):
+        state.results = None
+        state.raw_json = None
+
+    # Title
+    st.write("# Countries of the World")
 
-    # UI search bar and sidebar
-    st.write("# Haystack Demo")
+    # Sidebar
     st.sidebar.header("Options")
     top_k_reader = st.sidebar.slider("Max. number of answers", min_value=1, max_value=10, value=3, step=1)
-    top_k_retriever = st.sidebar.slider(
-        "Max. number of documents from retriever", min_value=1, max_value=10, value=3, step=1
-    )
+    top_k_retriever = st.sidebar.slider("Max. number of documents from retriever", min_value=1, max_value=10, value=3, step=1)
     eval_mode = st.sidebar.checkbox("Evaluation mode")
     debug = st.sidebar.checkbox("Show debug info")
 
-    st.sidebar.write("## File Upload:")
-    data_files = st.sidebar.file_uploader("", type=["pdf", "txt", "docx"], accept_multiple_files=True)
-    for data_file in data_files:
-        # Upload file
-        if data_file:
-            raw_json = upload_doc(data_file)
-            st.sidebar.write(raw_json)
-            if debug:
-                st.subheader("REST API JSON response")
-                st.sidebar.write(raw_json)
-
-    # load csv into pandas dataframe
+    # File upload block
+    if not DISABLE_FILE_UPLOAD:
+        st.sidebar.write("## File Upload:")
+        data_files = st.sidebar.file_uploader("", type=["pdf", "txt", "docx"], accept_multiple_files=True)
+        for data_file in data_files:
+            # Upload file
+            if data_file:
+                raw_json = upload_doc(data_file)
+                st.sidebar.write(str(data_file.name) + " &nbsp;&nbsp; ✅ ")
+                if debug:
+                    st.subheader("REST API JSON response")
+                    st.sidebar.write(raw_json)
+
+    st.sidebar.markdown("""
+    <style>
+        a {
+            text-decoration: none;
+        }
+        .haystack-footer {
+            text-align: center;
+        }
+        .haystack-footer h4 {
+            margin: 0.1rem; 
+            padding:0;
+        }
+        footer {
+            opacity: 0;
+        }
+    </style>
+    <div class="haystack-footer">
+        <hr />
+        <h4>Built with <a href="https://www.deepset.ai/haystack">Haystack 1.0</a></h4>
```

Comment:
I wonder whether we could reuse our `versionfromfile()` method from https://github.com/deepset-ai/haystack/blob/31e22012da1dfe298ee539a7344874f2f1013925/setup.py#L51
`_version: str = versionfromfile(here, "haystack", "_version.py")`

Reply:
As an alternative, maybe let's not mention the version number in the demo at all.

Reply:
Good idea actually! I will use `versionfromfile`

Reply:
Ok actually I'm not sure I can, because the UI and haystack are not necessarily installed in the same container... :sweat_smile: I'm gonna implement a real small endpoint on the rest api and use that, that's cleaner

Reply:
Or leave the version number out for now. It's not important from my point of view.